file_catalog/argbuilder.py

```diff
@@ -31,6 +30,35 @@ def build_start(kwargs: Dict[str, Any]) -> None:
             raise Exception("start is negative")
 
 
+def _resolve_path_args(kwargs: Dict[str, Any]) -> None:
+    """Resolve the path-type shortcut arguments by precedence.
+
+    Pop each key from `kwargs`, even if it's not used.
+    """
+    arg: Optional[Union[Dict[str, Any], str]] = None
+
+    # regex
+    if "path-regex" in kwargs:
+        arg = {"$regex": kwargs.pop("path-regex")}
+
+    # normal path
+    if "path" in kwargs:
+        arg = kwargs.pop("path")
+    if "logical_name" in kwargs:
+        arg = kwargs.pop("logical_name")
+
+    # directory & filename
+    if "directory" in kwargs or "filename" in kwargs:
+        if not (dpath := kwargs.pop("directory", "").rstrip("/")):
+            dpath = r".*"
+        if not (fname := kwargs.pop("filename", "").lstrip("/")):
+            fname = r".*"
+        arg = {"$regex": rf"^{dpath}((/)|(/.*/)){fname}$"}
```

Comment:
I think you need to escape the slashes.

I'm not a regex expert, but wondering if this would be better:
```
^{dpath}\/(.*\/)*{fname}$
```
It has only a single match group.

Reply:
"`/`"s aren't special characters, at least not in raw strings. `^{dpath}/(.*/)?{fname}$` would work, but I'm not sure it's going to be much of an improvement performance wise

Reply:
I do like the look of `^{dpath}/(.*/)?{fname}$` so I'm going to use that

Reply:
> "`/`"s aren't special characters, at least not in raw strings. `^{dpath}/(.*/)?{fname}$` would work, but I'm not sure it's going to be much of an improvement performance wise

I know it's not a special character in a string.  Just wondering how mongo is going to parse that.  I gues testing will find out.

Reply:
ahhh, so far so good

Reply:
https://stackoverflow.com/a/46616907/13156561